service-base/src/main/java/org/eclipse/hono/service/management/credentials/CommonSecret.java

```diff
@@ -55,6 +58,21 @@ public CommonSecret setEnabled(final Boolean enabled) {
         return this;
     }
 
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Sets the ID of the secret. This id may be assigned by the device registry.
```

Comment:
what is the uniqueness scope of this ID? I assume it is just the containing Credentials instance, right? However, IMHO we should document this here.

Reply:
I updated the description.

Reply:
Can these IDs be assigned by a client, i.e. when registering the credentials? If so, do we check that the IDs are unique within the credentials?